mypy/messages.py

```diff
@@ -998,6 +999,184 @@ def untyped_decorated_function(self, typ: Type, context: Context) -> None:
     def typed_function_untyped_decorator(self, func_name: str, context: Context) -> None:
         self.fail('Untyped decorator makes function "{}" untyped'.format(func_name), context)
 
+    def bad_proto_variance(self, actual: int, tvar_name: str, expected: int,
+                           context: Context) -> None:
+        msg = capitalize("{} type variable '{}' used in protocol where"
+                         " {} one is expected".format(variance_string(actual),
+                                                      tvar_name,
+                                                      variance_string(expected)))
+        self.fail(msg, context)
+
+    def concrete_only_assign(self, typ: Type, context: Context) -> None:
+        self.fail("Can only assign concrete classes to a variable of type '{}'"
+                  .format(self.format(typ)), context)
+
+    def concrete_only_call(self, typ: Type, context: Context) -> None:
+        self.fail("Only concrete class can be given where '{}' is expected"
+                  .format(self.format(typ)), context)
+
+    def note_call(self, subtype: Type, call: Type, context: Context) -> None:
+        self.note("'{}.__call__' has type '{}'".format(strip_quotes(self.format(subtype)),
+                                                       self.format(call)), context)
+
+    def report_protocol_problems(self, subtype: Union[Instance, TupleType, TypedDictType],
+                                 supertype: Instance, context: Context) -> None:
+        """Report possible protocol conflicts between 'subtype' and 'supertype'.
+        This includes missing members, incompatible types, and incompatible
```

Comment:
Style nit: Add empty line after the first sentence. Prefer to have multi-line docstrings formatted like this in general:

```py
def f(...) -> ...:
    """This is a single-line description.

    Longer description after an empty line ...
    """
```